### ISSUE TITLE
Make plugins support dataframes.

### ISSUE THREAD

**Issue by Berrysoft** [nushell/nushell#8100]
### Related problem

I have found that nushell supports [dataframes](https://www.nushell.sh/book/dataframes.html), and I would like to write plugins that deal with them. I would like to write plugins that read from other data files and output dataframes to nushell.

### Describe the solution you'd like

Support `PipelineData` in `Plugin`, both input and output.

As to the case of output, we need a new protocol, that is, a "chunked" list. A plugin prints values in chunks, and the shell parses the values in chunks. Thus we can make a lazy-evaluated list, and can deal with large data files.

**Comment by fdncred** [comment]
We've talked about this for a while. We'd love to have this support.

**Comment by sholderbach** [comment]
Since @devyn 's #11911 Nushell and a plugin can communicate bidirectionally in a streaming way.

Currently @ayax79 is trying to port the polars dataframe integration to a plugin #12220 (as polars itself is not yet stable in its semantics, making nushell stable with it is harder, and distribution benefits)

As things are right now the `PluginCustomValue`s are privy to each plugin so if you want to share data in the dataframe type of the future `polars` plugin, this would currently still have to go through a nushell `Value`/`PipelineData` intermediary.

**Comment by devyn** [comment]
Yes, I'm currently considering something along the lines of a `ForeignCustomValue` that can go to other plugins, and the ability to call other commands in the engine from the plugin (which would be the way you'd interact with the dataframes).

You would, unfortunately, be limited to their command interface for operations. There just isn't really a way to get a dataframe natively into your process space without trying to share a file or something like that. You'd be better off using polars as a library within your own plugin if you want to do anything too complicated I think.
